test/e2e/search_test.go

```diff
@@ -711,4 +717,197 @@ var _ = ginkgo.Describe("[karmada-search] karmada search testing", ginkgo.Ordere
 			})
 		})
 	})
+
+	// test when cluster joined, updated, karmada will reconcile search cache
+	ginkgo.Describe("reconcile ResourceRegistry when clusters joined, updated", ginkgo.Ordered, func() {
+		ginkgo.Context("when cluster joined", func() {
+			var (
+				clusterName    string
+				homeDir        string
+				kubeConfigPath string
+				controlPlane   string
+				clusterContext string
+				f              cmdutil.Factory
+
+				rrName           string
+				rr               *searchv1alpha1.ResourceRegistry
+				resourceSelector searchv1alpha1.ResourceSelector
+			)
+
+			ginkgo.BeforeAll(func() {
+				clusterName = "member-e2e-" + rand.String(3)
+				homeDir = os.Getenv("HOME")
+				kubeConfigPath = fmt.Sprintf("%s/.kube/%s.config", homeDir, clusterName)
+				controlPlane = fmt.Sprintf("%s-control-plane", clusterName)
+				clusterContext = fmt.Sprintf("kind-%s", clusterName)
+				defaultConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag().WithDiscoveryBurst(300).WithDiscoveryQPS(50.0)
+				defaultConfigFlags.Context = &karmadaContext
+				f = cmdutil.NewFactory(defaultConfigFlags)
+
+				// create a cluster
+				ginkgo.By(fmt.Sprintf("Creating cluster: %s", clusterName), func() {
+					err := createCluster(clusterName, kubeConfigPath, controlPlane, clusterContext)
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+				})
+
+				// create a ResourceRegistry with a non joined cluster name
+				rrName = "rr-" + rand.String(RandomStrLength)
```

Comment:
Put the `rr-` prefix into the constant block in suite_test.go file will be better.